pkg/config/apm.go

```diff
@@ -71,6 +71,9 @@ func setupAPM(config Config) {
 	config.BindEnvAndSetDefault("apm_config.remote_tagger", true, "DD_APM_REMOTE_TAGGER")                                                     //nolint:errcheck
 	config.BindEnvAndSetDefault("apm_config.peer_service_aggregation", false, "DD_APM_PEER_SERVICE_AGGREGATION")                              //nolint:errcheck
 	config.BindEnvAndSetDefault("apm_config.compute_stats_by_span_kind", false, "DD_APM_COMPUTE_STATS_BY_SPAN_KIND")                          //nolint:errcheck
+	config.BindEnvAndSetDefault("apm_config.instrumentation.enabled", false, "DD_APM_INSTRUMENTATION_ENABLED")
```

Comment:
Consider adding a test in `comp/trace/config/config_test.go` for all of tehse.

Reply:
Thanks for review @katiehockman! If I understand it correctly, the tests in `comp/trace/config/config_test.go` are validating configuration can be deserialized into AgentConfig object, but I've not added new fields for Instrumentation in AgentConfig. I don't quite understand how AgentConfig is used across the agent, but Cluster Agent doesn't seem to operate with it at all.